### PR TITLE
[@types/eonasdan-bootstrap-datetimepicker] Allow call methods

### PR DESCRIPTION
**Pull request by JoniJnm**
Allow call datetimepicker methods like show, destroy, etc.

- https://github.com/Eonasdan/bootstrap-datetimepicker/blob/master/src/js/bootstrap-datetimepicker.js#L2413
- http://eonasdan.github.io/bootstrap-datetimepicker/Functions/

### CODE REVIEW

**Comment by ToastHawaii** [comment]
@JoniJnm If you want. You can explizit specify the methods eg like this

/** description */
datetimepicker("destroy"): any;

/** description */
datetimepicker("hide"): JQuery;

**Comment by JoniJnm** [comment]
There are a lot of functions :(

File: types/eonasdan-bootstrap-datetimepicker/index.d.ts [hunk 1]
```diff
@@ -1209,6 +1209,7 @@ declare global {
     interface JQuery {
         datetimepicker(): JQuery;
         datetimepicker(options: SetOptions): JQuery;
+        datetimepicker(methodName: string, ...args: any[]): JQuery | any;
```
Comment (andy-ms) [inline types/eonasdan-bootstrap-datetimepicker/index.d.ts]: `JQuery | any` is just `any`.

**Review by ghost — COMMENTED** [review]
Does this take the name of a method in `Datetimepicker` and its arguments? If we had something like Microsoft/TypeScript#5453 we might have a way of representing that, until then, `any[]` may be the best we can do without copying all those definitions over.

File: types/eonasdan-bootstrap-datetimepicker/index.d.ts (hunk 1 above)
Comment (weswigham) [inline types/eonasdan-bootstrap-datetimepicker/index.d.ts]: Add
```ts
datetimepicker(methodName: "destroy"): JQuery;
datetimepicker(methodName: "hide"): JQuery;
datetimepicker(methodName: "show"): JQuery;
datetimepicker(methodName: "toggle"): JQuery;
```
Above this, then this line is fine - this should let chaining work with type inference, but still allow calling arbitrary methods.

**Review by weswigham — CHANGES_REQUESTED** [review]
Just these overloads should be fine, since most people care more about chaining than anything else (and these are the only methods that chain).

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed reviewer comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**